x-pack/plugins/monitoring/public/application/pages/no_data/no_data_page.tsx

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useContext, useState } from 'react';
+import { Redirect } from 'react-router-dom';
+
+import { i18n } from '@kbn/i18n';
+// @ts-ignore
+import { NoData } from '../../../components/no_data';
+import { PageTemplate } from '../page_template';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+import { CODE_PATH_LICENSE, STANDALONE_CLUSTER_CLUSTER_UUID } from '../../../../common/constants';
+import { Legacy } from '../../../legacy_shims';
+import { Enabler } from './enabler';
+import { BreadcrumbContainer } from '../../hooks/use_breadcrumbs';
+import { initSetupModeState } from '../../setup_mode/setup_mode';
+import { GlobalStateContext } from '../../global_state_context';
+
+const CODE_PATHS = [CODE_PATH_LICENSE];
+
+interface NoDataPageSetupDeps {
+  http: any;
+  data: any;
+}
+
+interface SettingsChecker {
+  message: string;
+  api: string;
+  next?: SettingsChecker;
+}
+
+const clusterCheckers: SettingsChecker[] = [
+  {
+    message: 'Checking cluster settings API on production cluster',
+    api: '../api/monitoring/v1/elasticsearch_settings/check/cluster',
+  },
+  {
+    message: 'Checking nodes settings API on production cluster',
+    api: '../api/monitoring/v1/elasticsearch_settings/check/nodes',
+  },
+];
+
+export const NoDataPage = () => {
+  const title = i18n.translate('xpack.monitoring.noData.routeTitle', {
+    defaultMessage: 'Setup Monitoring',
+  });
+
+  const { services } = useKibana<NoDataPageSetupDeps>();
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const [model, setModel] = useState({
+    errors: [], // errors can happen from trying to check or set ES settings
+    checkMessage: null, // message to show while waiting for api response
+    isLoading: true, // flag for in-progress state of checking for no data reason
+    isCollectionEnabledUpdating: false, // flags to indicate whether to show a spinner while waiting for ajax
+    isCollectionEnabledUpdated: false,
+    isCollectionIntervalUpdating: false,
+    isCollectionIntervalUpdated: false,
+  } as any);
```

Comment:
Don't think you need to change it for this PR, but usually you would want these to all be separate variables. For example: `const [isLoading, setIsLoading] = useState(true)`. You usually want to group things in objects if they are updated at the same time.

Reply:
Yeah, I was wondering about that. Thanks for the cross check.

The no data flow in angular seemed to contain a lot of `updateModel` type code so I thought this approach might be better to keep the diff smaller even it's not great practice.